core/templates/pages/contributor-dashboard-page/services/contribution-and-review-backend-api.service.ts

```diff
@@ -85,48 +86,64 @@ export class ContributionAndReviewBackendApiService {
   ) {}
 
   async fetchSuggestionsAsync(
-      fetchType: string
+      fetchType: string,
+      limit: number,
+      offset: number
   ): Promise<FetchSuggestionsResponse> {
     if (fetchType === this.SUBMITTED_QUESTION_SUGGESTIONS) {
-      return this.fetchSubmittedSuggestionsAsync('skill', 'add_question');
+      return this.fetchSubmittedSuggestionsAsync(
+        'skill', 'add_question', limit, offset);
     }
     if (fetchType === this.SUBMITTED_TRANSLATION_SUGGESTIONS) {
       return this.fetchSubmittedSuggestionsAsync(
-        'exploration', 'translate_content');
+        'exploration', 'translate_content', limit, offset);
     }
     if (fetchType === this.REVIEWABLE_QUESTION_SUGGESTIONS) {
-      return this.fetchReviewableSuggestionsAsync('skill', 'add_question');
+      return this.fetchReviewableSuggestionsAsync(
+        'skill', 'add_question', limit, offset);
     }
     if (fetchType === this.REVIEWABLE_TRANSLATION_SUGGESTIONS) {
       return this.fetchReviewableSuggestionsAsync(
-        'exploration', 'translate_content');
+        'exploration', 'translate_content', limit, offset);
     }
   }
 
   async fetchSubmittedSuggestionsAsync(
       targetType: string,
-      suggestionType: string
+      suggestionType: string,
+      limit: number,
+      offset: number
   ): Promise<FetchSuggestionsResponse> {
     const url = this.urlInterpolationService.interpolateUrl(
       this.SUBMITTED_SUGGESTION_LIST_HANDLER_URL, {
         target_type: targetType,
         suggestion_type: suggestionType
       }
     );
-    return this.http.get<FetchSuggestionsResponse>(url).toPromise();
+    const params = {
+      limit: limit.toString(),
```

Comment:
Shouldn't the toString() conversion be sorted out in the get() (as the more general nexus for all such calls)?

Reply:
The `params` object in the get() call expects the param values to be of type `string`, which is why I converted here. I think we would see a compiler issue otherwise?

Reply:
Ah ok. Hm, OK, I think we can leave it as-is.